MF-1329 - Add Python SDK

fbugarski opened the pull request:
Pull request title should be `MF-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
This pull request adds Python SDK.

### Which issue(s) does this PR fix/relate to?
Resolves #1329 

### List any changes that modify/break current functionality
Does not affect any current functionality.  

### Have you included tests for your changes?
Tests are not currently included, they will be added subsequently.

### Did you document any new/modified functionality?
Documentation will be added.

### Notes

fbugarski committed:
Add initial python SDK

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Update users SDK function

Added functions that implement users Mainflux
service API calls.

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Add things and channels

This commit adds SDK functions for
the things and channels microservices.

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

manuio commented on pkg/sdk/python/channels.py:
```diff
@@ -0,0 +1,32 @@
+    return resp
+
+def Channels(params, token):
+    url = "http://localhost/channels"+'?'+'offset='+params['offset']+'&'+'limit='+params['limit']+'&'+'connected='+params['connected']
```
What if `params['offset']` is empty?

fbugarski committed:
Add message and version

This commit adds message and version
SDK functions. Additionally, it
corrects things and channels.

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Add SDK class

This commit adds classes for all
modules and init module in SDK class.

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Pass url parameters

This commit adds configurable url params
for domain classes.

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Correct functions

This commit corrects function names.

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

manuio commented on pkg/sdk/python/channels.py:
```diff
@@ -20,13 +20,13 @@ def channels(self, query_params, token):
         return resp.json()
 
     def channel_by_thing(self, thing_id, query_params, token):
```
This returns a list of channels so better call it `channels_by_thing` or `list_chanels_by_thing`

fbugarski committed:
Fixed functions comments

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
ADDED status, error and msg

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Added value, error status and error msg

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Changes

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Add groups

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Add errors module

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Fixed query

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Fix groups

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Delete sdk.py

fbugarski committed:
Merge branch 'master' into pysdk

fbugarski committed:
Merge branch 'master' into pysdk

fbugarski committed:
Added tests

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

fbugarski committed:
Added tests and pkg

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

drasko commented on pkg/sdk/python/LICENSE.md:
```diff
@@ -0,0 +1,9 @@
+The MIT License (MIT)
```
No need for this license, we already have a license in the top-level Mainflux repo

drasko commented on pkg/sdk/python/tests/test_version.py:
```diff
@@ -0,0 +1,12 @@
+import json
+
+s = sdk.SDK()
+
```
Kill these whitelines

drasko commented on pkg/sdk/python/setup.cfg:
```diff
@@ -0,0 +1,24 @@
+
+[options]
+package_dir =
+    = src
```
This is strange - why not in the same line?

drasko commented on pkg/sdk/python/functions/examples.py:
```diff
@@ -0,0 +1,233 @@
+
+sdk = SDK()
+
+# '''To start working with the Mainflux system, you need to create a user account'''
```
This will be needed to be uncommented in the end

fbugarski committed:
fixing examples

drasko commented on pkg/sdk/python/setup.py:
```diff
@@ -0,0 +1,26 @@
+    long_description = fh.read()
+
+setuptools.setup(
+    name="example-pkg-fbugarski",
```
Name of the SDK is probably `mainflux`

fbugarski committed:
Changed files organization

Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

manuio commented on pkg/sdk/python/setup.py:
```diff
@@ -17,7 +17,6 @@
     },
     classifiers=[
         "Programming Language :: Python :: 3.8.5",
-        "License :: OSI Approved :: MIT License",
```
Maybe better to define the licence?

manuio commented on pkg/sdk/python/functions/examples.py:
```diff
@@ -1,24 +1,25 @@
+from functions import groups
 import requests
-from sdk import SDK
+from functions import SDK
```
This sounds too generic as naming
